content/posts/gitops-series-part-1.md

```diff
@@ -0,0 +1,618 @@
+
+![GitOps Trend](/img/gitops-series/part-1/trend.png)
+
+If you haven't already done so, I'd recommend reading that Weaveworks post since it is always good to understand the origination of a concept. But simply put, GitOps is a way for you to manage your operations from a source code repo. With GitOps you won't be doing any manual steps when you want to change something. On a commit to your master branch, a job will get kicked off and will make the necessary changes.
+
+## Why would you want this?
+
+If you have any experience on an operations team with little or no automation, you no doubt know the frustration of manual configuration and snowflake servers. It this sort of environment it is easy to quickly get overwhelmed. And when things go wrong, they can spin out of control.
+
+As you move to infrastructure-as-code by using configuration management tools you're able to get away from most of that headache. Now that you have code which describes your desired environment state you have an easy way to manage and maintain the environment. When you need to change something, submit a pull request and have someone review it. Once the change is merged, go ahead and run the automation and the change will propogate. Should something disastrous happen to your environment, you can get your environment back up in no time by rerunning the automation.
+
+But you are still missing something if these commits to master don't automatically kick off a job to make the change. You should always want your environment to 100% match the configuration in your repo. If the automation isn't automatically run, you will drift away from the target state. 
+
+I've personally been guilty of putting off running automation due to fear of something breaking, and I know I'm not alone in this. Knowing that merging your pull request is going to trigger a job makes you more careful in your review of pull requests but also gives you confidence in knowing that your environment is always up-to-date.
+
+
+## Objective of this series
+
+To explore the idea of GitOps I am writing a 3-part series where we'll be building out a fully-functional GitOps process. 
+
+In this first part we will take a look at building out the infrastructure automation piece. This will involve provisioning a Kubernetes cluster, setting up certificates and DNS, and more. From here we will fork into two different directions.
+
+In the second part we will add the automation of our [CloudBees Jenkins Distribution](https://www.cloudbees.com/products/cloudbees-jenkins-distribution). This will include plugin management, configuration, and more.
+
+In the final part we will look at [CloudBees Core](https://www.cloudbees.com/products/cloudbees-core) and some cool stuff we can do there with custom Operations Center and Managed Master images.
```

Comment:
I think this would sound better if you change "some cool stuff we can do there with custom..." to "some cool stuff we can do with custom..."

Reply:
Thanks for catching these, I've fixed those issues.